Added documentation for running tests (#477)
10up/plugin-check: docs/run-unit-tests.md

[Back to overview](./README.md)

> 

Reviewer: This file should be renamed to `running-unit-tests.md` from `run-unit-tests.md`

# Technical Overview

> To ensure that all checks are working correctly, the Team has developed Unit test to run the checks whether it pass or fail, to ensure the consistency of the rest of checks and functions. The tests are based on [wp-env](https://make.wordpress.org/core/2020/03/03/wp-env-simple-local-environments-for-wordpress/), so you could run a clean installation to run all tests.

Reviewer 2: ```suggestion
To ensure that all checks are working correctly, the team is maintaining a set of unit tests. The tests setup is based on [wp-env](https://make.wordpress.org/core/2020/03/03/wp-env-simple-local-environments-for-wordpress/), so you could run a clean installation to run all tests.
```

Follow these instructions to configure and run tests:

1. You would need npm installed in your computer.
2. Ensure that you have installed all npm dependencies with `npm ci`.
3. You need to have Docker installed and running wp-env environment `npm run wp-env start`.
4. Run tests with npm command `npm run test-php`.

> The full test suite is run against PRs as a GitHub action ( [example](https://github.com/WordPress/plugin-check/actions/runs/9660204610) ) so tests can be run against all supported environments. Making sure those pass are the important thing. Being able to run them locally is meant to help developers while working on or debugging tests, prior to submitting their code for review.

Reviewer 2: ```suggestion
The full test suite is run against PRs as a GitHub action ([example](https://github.com/WordPress/plugin-check/actions/runs/9660204610)) so tests can be run against all supported environments. Passing tests is a requirement for merging PRs. Being able to run them locally is meant to help developers while working on or debugging tests, prior to submitting their code for review.
```

## Where to find folder tests

We have this structure for the tests folder:

- `tests/phpunit` Unit tests for the plugin.
  - `tests/phpunit/tests/` All PHPUnit tests that run as part of the suite.  
  - `tests/phpunit/testdata/` Example classes or plugins that can be used in tests.  
  - `tests/phpunit/utils/` Shared helpers that can be used across different test classes.  

> Ensure that you create your Check and test unit at same time.

Reviewer 2: ```suggestion
Ensure that you create your check and unit test at same time.
```

> [Back to overview](./README.md)

Reviewer 3: Rename this file to match the updates suggested in the README.md table of contents.

Author: it's already updated i believe

# Technical Overview

> To ensure that all Checks are working correctly, the Team has developed Unit test to run the checks whether it pass or fail, to ensure the consistency of the rest of checks and functions. The tests are based on [wp-env](https://make.wordpress.org/core/2020/03/03/wp-env-simple-local-environments-for-wordpress/), so you could run a clean installation to run all tests.

Reviewer 3: ```suggestion
To ensure that all checks are working correctly, the Team has developed Unit test to run the checks whether it pass or fail, to ensure the consistency of the rest of checks and functions. The tests are based on [wp-env](https://make.wordpress.org/core/2020/03/03/wp-env-simple-local-environments-for-wordpress/), so you could run a clean installation to run all tests.
```

Follow these instructions to configure and run tests:

1. You would need npm installed in your computer.
2. Ensure that you have installed all npm dependencies with `npm ci`.
3. You need to have Docker installed and running wp-env environment `npm run wp-env start`.
4. Run tests with npm command `npm run test-php`.

> If it does not fail, your Pull request is ready to be commited to the repository.

Reviewer 4: ```suggestion
If it does not fail, your Pull request is ready to be committed to the repository.
```

> If it does not fail, your Pull request is ready to be committed to the repository.

Reviewer 3: The full test suite is run against PRs as a GitHub action ([example](https://github.com/WordPress/plugin-check/actions/runs/9660204610)) so tests can be run against all supported environments. Making sure those pass are the important thing. Being able to run them locally is meant to help developers while working on or debugging tests, prior to submitting their code for review.

## Where to find folder tests

We have this structure for the tests folder:

- tests/behat

> - tests/phpstan

Reviewer 3: These aren't unit tests, and should likely have their own documentation rather than be included here.

Author: removed!

> - tests/phpunit Unit tests for the plugin.
>   - tests/phpunit/tests/Checker/Checks All tests to run inside the plugin.
>   - tests/phpunit/testdata/plugins You will find all data required to run unit tests, organized by folders.

Reviewer 3: I don't think we should single out a specific sets of tests and exclude others. All of them need to run and pass. Also, we can explain why the `/utils` directory is there.

```suggestion
- `tests/phpunit` Unit tests for the plugin.
  - `tests/phpunit/tests/` All PHPUnit tests that run as part of the suite.
  - `tests/phpunit/testdata/` Example classes or plugins that can be used in tests.
  - `tests/phpunit/utils/` Shared helpers that can be used across different test classes.
```

Author: ok!
